cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/StageOperationsValidator.java

```diff
@@ -183,6 +187,8 @@ void validate() {
           case WRITE:
             // this will not happen since a write operation will not have output fields, and thus no field will
             // appear in the unusedOutput
+            LOG.warn("The write operation {} should never generate the output field {}, something messed up in the " +
```

Comment:
Can we make the message a little better? Like: "Field Level Lineage encountered an unexpected situation: Write operations should not have output fields, but operation {} declared output field {}. This may cause recording of field level lineage to fail."